tests/rptest/tests/license_enforcement_test.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import re
+
+from ducktape.mark import matrix
+
+from rptest.services.cluster import cluster
+from rptest.clients.rpk import RpkTool
+from rptest.services.redpanda import LoggingConfig
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda_installer import RedpandaInstaller
+
+
+class LicenseEnforcementTest(RedpandaTest):
+    # Disable log checks because it is not very useful when we expect to crash nodes on license upgrades
+    LOG_ALLOW_LIST = [re.compile(".*")]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args,
+                         num_brokers=5,
+                         log_config=LoggingConfig('info',
+                                                  logger_levels={
+                                                      'cluster': 'debug',
+                                                      'features': 'debug'
+                                                  }),
+                         **kwargs)
+
+        self.rpk = RpkTool(self.redpanda)
+
+    def setUp(self):
+        # start the nodes manually
+        pass
+
+    @cluster(num_nodes=5, log_allow_list=LOG_ALLOW_LIST)
+    @matrix(
+        clean_node_before_recovery=[False, True],
+        clean_node_after_recovery=[False, True],
+    )
+    def test_license_enforcement(self, clean_node_before_recovery,
+                                 clean_node_after_recovery):
+        installer = self.redpanda._installer
+
+        prev_version = installer.highest_from_prior_feature_version(
+            RedpandaInstaller.HEAD)
+        latest_version = installer.head_version()
+        self.logger.info(
+            f"Testing with versions: {prev_version=} {latest_version=}")
```

Comment:
chore/discussion: nothing wrong here, but triggered this thought: I think once we branch the release (say v24.3.x), we should update the feature table as the next commit to dev.  I think this should be policy in core.  Thoughts @dotnwat @mmaslankaprv ?